Adding an animation to the workday waiver holidays table when switching contents

araujoarthur0 opened the pull request:
#### Related issue
Closes #1009

Before:
![holidays](https://github.com/thamara/time-to-leave/assets/37311270/e12526c7-a0cb-4d8d-91e6-4b5c2d1471e1)

After:
![holidays2](https://github.com/thamara/time-to-leave/assets/37311270/ca1c5ada-99c0-429f-83dd-c9221f579d80)

Also limited the min width of the date column so it stops breaking when the holiday names are too big.

araujoarthur0 commented:
The gifs are awfully slow, but currently there's a split second in which the table is empty and the window just pops up and down.
Now it waits in time to paint the new table.

tupaschoal reviewed (approved):
Looks great!

tupaschoal commented:
Seems like the tests covering clearing the table need a look @araujoarthur0

tupaschoal commented:
@araujoarthur0 the tests are failing cause the rows are not immediately removed because of the fadeout, need to wait for the fadeout to happen or disable them before running the tests.

araujoarthur0 commented:
Seems the tests are failing on main as well

araujoarthur0 commented:
Turns out it was also failing for me in main because of a saved preference from another test. Restoring the default preferences at the beginning of the test fixed the second issue. Now back to the animation issue.

araujoarthur0 commented:
The removal wasn't working because $(this) does not work in a lambda like I did.
Fixing it returns the animation problems, so I'll rethink my approach.

araujoarthur0 commented:
Fixed the implementation by using a temporary table in which the html operations happen in the background, and finish by switching the two tables.

Also fixed it that choosing '--' as a country was not clearing the table, and updating the year was not updating the table holidays.

araujoarthur0 commented:
\changelog-updater

araujoarthur0 commented:
\changelog-update
Message: Enhancement [#1009]: Adding an animation to the workday waiver holidays table when the contents switch
